core/core_modules/admin.js

```diff
@@ -0,0 +1,176 @@
+﻿var config = require.once('../config.js'),
+
+    getHasPermission = function (config, userId, name, threadId, moduleName) {
+        if (!config.modules || !config.modules[moduleName] || config.modules[moduleName].length === 0) {
+            return true;
+        }
+
+        if (!config.users || Object.keys(config.users).length === 0) {
+            return false;
+        }
+
+        var user = config.users[userId] || config.users[name];
+        if (!user) {
+            return false;
+        }
+
+        var tId = null;
+        for (var thread in user) {
+            var test = new RegExp(thread);
+            if (test.test(threadId)) {
+                tId = thread;
+                break;
+            }
+        }
+        if (!tId || user[tId].length === 0) {
+            return false;
+        }
+
+        var common = user[tId].filter(function (p) {
+            return config.modules[moduleName].indexOf(p) !== -1;
+        });
+
+        return common.length >= 1;
+    },
+
+    modify = function (name, threadId, action, method) {
+        if (!exports.config.users) {
+            exports.config.users = {};
+        }
+
+        if (!exports.config.users[name]) {
+            exports.config.users[name] = {};
+        }
+
+        if (!exports.config.users[name][threadId]) {
+            exports.config.users[name][threadId] = [];
+        }
+
+        switch (method) {
+            case 'grant': {
+                if (!exports.config.users[name][threadId].includes(action)) {
+                    exports.config.users[name][threadId].push(action);
+                    return true;
+                }
+                return false;
+            }
+            case 'revoke': {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 8 space characters but found 12.](https://www.codacy.com/app/theprojectshop/Kassy/file/2587223488/issues/source?bid=3282987&fileBranchId=3292072#l57)